### DISCUSSION TITLE
Bonus room terrain object definitions

### DISCUSSION THREAD

**Discussion by nintyalex** [city41/smaghetti#120] [General]
In addition to what's in the [checkered floor section](https://github.com/city41/smaghetti/wiki/Checkered-terrain) of the wiki, I've found more objects that define that terrain:
Bank 08, id 4f is the left end of a one-tile-tall platform; bank 4e, id 2e, param (length) is its middle; and bank 4f, id 2e, param 01 is the right end.
Bank 42, id 2e, param (length) is a checkered ceiling.
Bank 45 seems to also be a right wall, like bank 4c; likewise bank 44 is a left wall, like bank 4b. Bank 41 is the upper left corner, and bank 43 is the upper right corner.
If bank 4c, id 2e with param 01, is a lupper left corner (floor-to-wall); bank 4b, id 2e with param 01 makes a upper right corner (floor-to-wall).
Bank 4a, id 2e is a lower left corner (wall-to-ceiling), while bank 49, id 2e is a lower right corner (wall-to-ceiling).
Bank 40 is a six-byte object, with id 2e, param (length), and param2 (height); it is another ceiling.

The graphics sets look to be [0 or 5, 0, 0, 2, 0, 0 or 1].

**Comment by city41** [comment]
Awesome, thanks! I just realized only I can edit the wiki. Do you want me to give you access? Happy to let you edit the wiki any way you see fit. To do that you'd be marked as a collaborator on smaghetti.

**Reply by city41** [reply]
Also since these are objects and not sprites, I don't think they care about the sprite graphic sets. Objects have object sets and object graphic sets they need, but a room can only have one object set and one object graphic set loaded, compared to the six graphic sets for sprites.

**Reply by nintyalex** [reply]
Yeah, sure.

Oh, ok. So they use object set 01.

**Reply by city41** [reply]
cool, I just invited you. The checkered floor I already extracted is available in a lot of object sets, they are [here](https://github.com/city41/smaghetti/blob/main/src/entities/CheckeredTerrain/objectSets.ts)
  
where the first number in an array is the object set, and the second number is the object graphic set. Most objects are in tons of sets like this, but some are only in one or two, like [ColorfulMetalBox](https://github.com/city41/smaghetti/blob/main/src/entities/ColorfulMetalBox.tsx#L58)  
  
So it's possible these new objects you found might also be in a lot of sets, which would be great.

**Reply by city41** [reply]
Or I should say, colorful metal box appears to only be in 1. I'm not convinced my scripts that figure out object sets are totally thorough. It's possible there are more sets, which if true, could possibly increase compatibility between entities in a room, which would be most excellent. That's what [this issue](https://github.com/city41/smaghetti/issues/34) is about.

**Reply by city41** [reply]
and while we're on this subject, `bgColor` in level settings also influences objects. For example it's the main thing that causes pipes to be different colors. I don't fully understand it though.  
  
I am thinking for my next focus in smaghetti will be disassembling and decompiling the game, and seeing if we can get better understandings of things that way.

**Comment by city41** [comment]
@nintyalex thanks for all the new reverse engineering in the wiki! I will try and get at least some of it in this weekend.
